modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/containers/GenerateNewKey/pages/SelectSubscription.jsx

```diff
@@ -170,12 +249,186 @@ const SelectSubscription = ({
 
 	const subscriptionTerms = useMemo(
 		() =>
-			generateFormValues?.subscriptionTerms?.filter(
-				(key) => key.productKey === selectedProductKey
+			generateFormValues?.subscriptionTerms?.filter((key) =>
+				state.id === 'renew'
+					? matchingProductKeys.includes(key.productKey)
+					: key.productKey === selectedProductKey
 			),
-		[generateFormValues?.subscriptionTerms, selectedProductKey]
+		[
+			generateFormValues?.subscriptionTerms,
+			selectedProductKey,
+			state.id,
+			matchingProductKeys,
+		]
 	);
 
+	const submitKey = useCallback(async () => {
+		const licenseEntryTypes = state.activationKeys.map((key) => {
+			return key.licenseEntryType;
+		});
+
+		const licenseEntryType =
+			licenseEntryTypes.includes('virtual-cluster') ||
+			licenseEntryTypes?.includes('oem') ||
+			licenseEntryTypes?.includes('Enterprise');
+
+		const nameOfSelectedProducts = [...new Set(licenseEntryTypes)]
```

Comment:
rename to `selectedProductNames`. its a little shorter and gets rid of the unnecessary `of` in the naming